pkg/config/parameters.go

```diff
@@ -51,19 +51,6 @@ func (g *GatewayParameters) Validate() error {
 		return nil
 	}
 
-	if len(g.Name) == 0 && len(g.Namespace) == 0 && len(g.ControllerName) == 0 {
-		return nil
-	}
-
-	if len(g.Name) == 0 {
```

Comment:
this should have been removed when the deprecation notice was added

Reply:
We ended up going slower than I anticipated on this unfortunately. This will be cleaned up in Contour v1.19.